api/entities.go

```diff
@@ -0,0 +1,83 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+type EntitiesService struct {
+	client *Client
+}
+
+type EntityType int
+
+const (
+	NoneEntityType EntityType = iota
+	MachineDetailsEntityType
+	UsersEntityType
+	ImagesEntityType
+)
+
+// EntityTypes is the list of available entity types
+var EntityTypes = map[EntityType]string{
+	NoneEntityType:           "None",
+	MachineDetailsEntityType: "MachineDetails",
+	UsersEntityType:          "Users",
+	ImagesEntityType:         "Images",
+}
+
+// Search expects the response and the search filters
+//
+// e.g.
+//
+//   var (
+//       response = &api.MachineDetailsEntityResponse{}
+//       now      = time.Now().UTC()
+//       before   = now.AddDate(0, 0, -7) // 7 days from ago
+//       filters  = api.SearchFilter{
+//           TimeFilter: &api.TimeFilter{
+//               StartTime: &before,
+//               EndTime:   &now,
+//           },
+//       }
+//   )
+//   lacework.V2.Entities.Search(response, filters)
+//
+func (svc *EntitiesService) Search(response interface{}, filters SearchFilter) error {
+	var apiPath string
+
+	switch response.(type) {
+	case *MachineDetailsEntityResponse:
+		apiPath = fmt.Sprintf(apiV2EntitiesSearch, EntityTypes[MachineDetailsEntityType])
+
```

Comment:
Is the spacing between case statements personal preference? If so perhaps we should add it to linter settings. I've no strong but I've seen this pop on PR's and usually recommend to not space out between cases.